packages/http-client/__tests__/proxy.test.ts

```diff
@@ -237,6 +237,31 @@ describe('proxy', () => {
     expect(_proxyConnects).toHaveLength(0)
   })
 
+  it('HttpClient bypasses proxy for loopback addresses (localhost, ::1, 127.*)', async () => {
+    // setup a server listening on localhost:8091
+    const server = http.createServer((request, response) => {
+      response.writeHead(200)
+      request.pipe(response)
+    })
+    server.listen(8091)
+    try {
+      process.env['http_proxy'] = _proxyUrl
+      const httpClient = new httpm.HttpClient()
+      let res = await httpClient.get('http://localhost:8091')
+      expect(res.message.statusCode).toBe(200)
+      res = await httpClient.get('http://127.0.0.1:8091')
+      expect(res.message.statusCode).toBe(200)
+
+      // no support for ipv6 for now
+      expect(httpClient.get('http://[::1]:8091')).rejects.toThrow()
```

Comment:
Nit: I think this assertion is not relevant to this test case (we are testing the behaviour of proxy for loopback address). Maybe it could be moved into separate test?

Reply:
Currently we don't support ipv6 in `httpClient.get('http://[::1]:8091')`. Once we do (aka `httpClient.get` no longer throws), I want this test to fail, because it means proxy bypass for ipv6 loopback adresses can now be tested and verified.